logic/strategy/src/one_hop_connection.rs

```diff
@@ -0,0 +1,358 @@
+    /// Ethereum addresses of the destination peers.
+    #[validate(length(min = 1))]
+    #[serde_as(as = "Vec<DisplayFromStr>")]
+    pub destination_peers: Vec<Address>,
+
+    /// A quality threshold between 0 and 1 used to determine whether the strategy should open channel with the peer.
+    #[validate(range(min = 0_f64, max = 1.0_f64))]
+    #[default = 0.5]
+    pub network_quality_threshold: f64,
+
+    /// Specifies a minimum version (in semver syntax) of the peer the strategy should open a channel to.
+    ///
+    /// Default is ">=2.0.0"
+    #[serde_as(as = "DisplayFromStr")]
+    #[default(">=2.0.0".parse().expect("should be valid default version"))]
+    pub minimum_peer_version: semver::VersionReq,
+
+    /// Minimum stake that a channel's balance must not go below.
+    ///
+    /// Default is 1 HOPR
+    #[serde_as(as = "DisplayFromStr")]
+    #[default(Balance::new_from_str("1000000000000000000", BalanceType::HOPR))]
+    pub min_stake_threshold: Balance,
+
+    /// Funding amount.
+    ///
+    /// Defaults to 10 HOPR.
+    #[serde_as(as = "DisplayFromStr")]
+    #[default(Balance::new_from_str("10000000000000000000", BalanceType::HOPR))]
+    pub funding_amount: Balance,
+}
+
+/// This strategy is designed to maintain 1-hop pathes to certain destination peers,
+/// where there is at least one relay node that is not part of the destination peer set.
+///
+/// There should be outgoing channels to the destination peers.
+///
+/// When there's no relay node that is not part of the destination peer set,
+/// the strategy should select a relay node and open an outgoing channel to it.
+/// Selection of this new relay peer should prioritize those that have incoming channels
+/// with the destination peers.
+///
+/// It choses channels to close among the channels that cannot be used for reaching
+/// the destination peers with 1-hop.
+/// The strategy only outgoing channels where the target has low network quality.
+pub struct OneHopConnectionStrategy<A, Db>
+where
+    Db: HoprDbAllOperations + Clone,
+    A: ChannelActions + Send + Sync,
+{
+    chain_actions: A,
+    db: Db,
+    cfg: OneHopConnectionStrategyConfig,
+}
+
+impl<A, Db> OneHopConnectionStrategy<A, Db>
+where
+    Db: HoprDbAllOperations + Clone,
+    A: ChannelActions + Send + Sync,
+{
+    pub fn new(cfg: OneHopConnectionStrategyConfig, db: Db, chain_actions: A) -> Self {
+        Self { chain_actions, db, cfg }
+    }
+
+    async fn get_peers_with_quality(&self) -> Result<HashMap<Address, f64>> {
+        Ok(self
+            .db
+            .get_network_peers(PeerSelector::default(), false)
+            .await?
+            .filter_map(|status| {
+                let version = status.peer_version.clone().and_then(|v| {
+                    semver::Version::from_str(&v)
+                        .ok() // Workaround for https://github.com/dtolnay/semver/issues/315
+                        .map(|v| Version::new(v.major, v.minor, v.patch))
+                });
```

Comment:
In 3.0.0 the version will likely not be directly available like this, for 2 reasons:
1. custom implementations of whatever on top of HOPR means there's not 1 single version of whatever is on the other side
2. it leaks information about the node itself, which should not be utilized

Reply:
In singapore this code can stay.